researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider2.java

```diff
@@ -0,0 +1,284 @@
+package org.sagebionetworks.bridge.researchstack;
+
+import android.content.Context;
+import android.support.annotation.AnyThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.researchstack.backbone.DataProvider;
+import org.researchstack.backbone.DataResponse;
+import org.researchstack.backbone.model.ConsentSignatureBody;
+import org.researchstack.backbone.model.SchedulesAndTasksModel;
+import org.researchstack.backbone.model.User;
+import org.researchstack.backbone.result.TaskResult;
+import org.researchstack.backbone.task.Task;
+import org.sagebionetworks.bridge.android.manager.BridgeManagerProvider;
+import org.sagebionetworks.bridge.android.manager.auth.AuthenticationManager;
+import org.sagebionetworks.bridge.rest.model.SharingScope;
+import org.sagebionetworks.bridge.rest.model.StudyParticipant;
+import org.sagebionetworks.bridge.rest.model.UserSessionInfo;
+
+import rx.Completable;
+import rx.Observable;
+import rx.Observer;
+import rx.functions.Action0;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sagebionetworks.bridge.researchstack.ApiUtils.SUCCESS_DATA_RESPONSE;
+
+/**
+ * Created by jyliu on 1/20/2017.
+ */
+@AnyThread
+public class BridgeDataProvider2 extends DataProvider {
```

Comment:
This is temporary.  I want to avoid blocking progress by breaking authentication so I am building this along side BridgeDataProvider.  This code is a work in progress and only simple happy cases have been tested.

My plan is for BridgeDataProvider (old code) to temporarily extend BridgeDataProvider2 (new code). This is to make it easier to run a mix of old and new code, e.g. commenting out methods in BridgeDataProvider, to allow BridgeDataProvider2's new code to be called, or in the old code, super.methodName() can be called so both old and new code are invoked (useful for things that should change state in both, like if the credentials are changed)